lib/HttpKernel/CacheWarmer/PimcoreCoreCacheWarmer.php

```diff
@@ -79,7 +79,9 @@ private function getClassesFromDirectory(string $dir, string $excludePattern, st
                 $className = preg_replace('@\.php$@', '', $className);
                 $className = str_replace(DIRECTORY_SEPARATOR, '\\', $className);
 
-                if (class_exists($className)) {
+                // include classes, interfaces and traits
+                // exclude invalid files like helper-functions
+                if (!preg_match('/[_.-]/', $className)) {
```

Comment:
@mattamon should we maybe check for a valid class name here, as described here: 
https://www.php.net/manual/en/language.oop5.basic.php#:~:text=The%20class%20name%20can%20be,_%5Cx80-%5Cxff%5D*%24%20. (plus `\` - I think that's missing in the regexp)

Reply:
Class names cannot contain `\`, because it's the namespace separator.

Reply:
But it's part of the string we check, right? Because it's the FQCN, not?

Reply:
Yes the `$className` is actually the FQCN

Reply:
If you want to check the FQCN, I recently stumbled across this crazy regex [in the Symfony code base](https://github.com/symfony/symfony/blob/f3a09b9a9957546118bf06d3d0570598afb76cb0/src/Symfony/Component/DependencyInjection/Attribute/AutowireIterator.php#L59C52-L59C204): 
`/(?(DEFINE)(?<cn>[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+))(?(DEFINE)(?<fqcn>(?&cn)(?:\\\\(?&cn))*+))^\??(?&fqcn)(?:(?:\|(?&fqcn))*+|(?:&(?&fqcn))*+)$/`

Reply:
@jdreesen wooooow, I would not even pretend to understand this regex :D 
But thank you very much! @brusch wdyt? 
Does it make sense to check if or are we sure, that our internal classes meet the expectations anyway?

Reply:
Me neither, so I tried to understand what it does with the help of the explanation box of https://regex101.com/ :grin:

Reply:
@mattamon I'd go for the simple solution for now (keep it as it is), if we see issues we can refine it anyway 😎